pkg/ddevapp/config.go

```diff
@@ -874,26 +872,30 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg:
 
 	// For webimage, update to latest composer.
 	if strings.Contains(fullpath, "webimageBuild") {
-		// If composerVersion is set,
-		// run composer self-update to the version (or --1 or --2)
-		// defaults to "2" even if ""
+		// Version to run composer self-update to the version
 		var composerSelfUpdateArg string
-		switch composerVersion {
-		case "1":
-			composerSelfUpdateArg = "--1"
-		case "":
-			fallthrough
-		case "2":
-			composerSelfUpdateArg = "--2"
-		default:
-			composerSelfUpdateArg = composerVersion
-		}
+
+		// Remove leading and trailing spaces
+		composerSelfUpdateArg = strings.TrimSpace(composerVersion)
 
 		// Composer v2 is default
+		if composerSelfUpdateArg == "" {
+			composerSelfUpdateArg = "2"
+		}
+
+		// Major and minor versions have to be provided as option so add '--' prefix.
+		// E.g. a major version can be 1 or 2, a minor version 2.2 or 2.1 etc.
+		if strings.Count(composerVersion, ".") < 2 {
+			composerSelfUpdateArg = "--" + composerSelfUpdateArg
+		}
+
 		// Try composer self-update twice because of troubles with composer downloads
 		// breaking testing.
+		// First of all Composer is updated to latest stable release to ensure
+		// new options of the self-update command can be used properly e.g.
+		// selecting a branch instead of a major version only.
 		contents = contents + fmt.Sprintf(`
-RUN export XDEBUG_MODE=off && ( composer self-update %s || composer self-update %s || true )
+RUN export XDEBUG_MODE=off; composer self-update --stable || composer self-update --stable; composer self-update %s || composer self-update %s || true
```

Comment:
```suggestion
RUN export XDEBUG_MODE=off; composer self-update --stable || composer self-update --stable || true; composer self-update %s || composer self-update %s || true
```

We need the `|| true` here on the first update because errors cause failure, and this can fail when no internet or when getcomposer is misbehaving, etc.

Reply:
Do we even need the initial `self-update --stable` here, which has some additional cost? I know that it would have been an issue before v1.19.2, but since v1.19.2 now ships with a composer version that has all these features... do we need to do this stuff?

Reply:
The idea behind this is to avoid similar issues in the future when Composer introduces new version features again. Having the latest stable version installed first makes sure we can use everything Composer supports at this time. And it's a one time thing on the first image creation. As we saw Docker build cache is very effective here and caches also between various images so I really don't think this has a big impact but helps to avoid issues.

Reply:
Maybe we should do some tests again with the Docker build cache and the new `debug refresh` command to see how the behavior is at the moment. Needs ofc a new Composer release...